Recorder/OutputType.cpp

```diff
@@ -463,169 +131,15 @@ int to_index(const OutputType config) {
-    if(OutputType::DF5 == L) return "DF";
-    if(OutputType::DF6 == L) return "DF";
-    if(OutputType::DM1 == L) return "DF";
-    if(OutputType::DM2 == L) return "DF";
-    if(OutputType::DM3 == L) return "DF";
-    if(OutputType::IF == L) return "IF";
-    if(OutputType::IF1 == L) return "IF";
-    if(OutputType::IF2 == L) return "IF";
-    if(OutputType::IF3 == L) return "IF";
-    if(OutputType::IF4 == L) return "IF";
-    if(OutputType::IF5 == L) return "IF";
-    if(OutputType::IF6 == L) return "IF";
-    if(OutputType::IM1 == L) return "IF";
-    if(OutputType::IM2 == L) return "IF";
-    if(OutputType::IM3 == L) return "IF";
-    if(OutputType::GDF == L) return "GDF";
-    if(OutputType::GDF1 == L) return "GDF";
-    if(OutputType::GDF2 == L) return "GDF";
-    if(OutputType::GDF3 == L) return "GDF";
-    if(OutputType::GDF4 == L) return "GDF";
-    if(OutputType::GDF5 == L) return "GDF";
-    if(OutputType::GDF6 == L) return "GDF";
-    if(OutputType::GDM1 == L) return "GDM";
-    if(OutputType::GDM2 == L) return "GDM";
-    if(OutputType::GDM3 == L) return "GDM";
-    if(OutputType::GIF == L) return "GIF";
-    if(OutputType::GIF1 == L) return "GIF";
-    if(OutputType::GIF2 == L) return "GIF";
-    if(OutputType::GIF3 == L) return "GIF";
-    if(OutputType::GIF4 == L) return "GIF";
-    if(OutputType::GIF5 == L) return "GIF";
-    if(OutputType::GIF6 == L) return "GIF";
-    if(OutputType::GIM1 == L) return "GIF";
-    if(OutputType::GIM2 == L) return "GIF";
-    if(OutputType::GIM3 == L) return "GIF";
-    if(OutputType::HYDRO == L) return "HYDRO";
-    if(OutputType::MISES == L) return "MISES";
-    if(OutputType::EEQ == L) return "EEQ";
-    if(OutputType::EEEQ == L) return "EEEQ";
-    if(OutputType::PEEQ == L) return "PEEQ";
-    if(OutputType::KE == L) return "KE";
-    if(OutputType::SE == L) return "SE";
-    if(OutputType::VE == L) return "VE";
-    if(OutputType::NVE == L) return "NVE";
-    if(OutputType::K == L) return "K";
-    if(OutputType::M == L) return "M";
-    if(OutputType::ED == L) return "ED";
-    if(OutputType::VD == L) return "VD";
-    if(OutputType::SD == L) return "SD";
-    if(OutputType::ES == L) return "ES";
-    if(OutputType::VS == L) return "VS";
-    if(OutputType::SS == L) return "SS";
-    if(OutputType::DAMAGE == L) return "DAMAGE";
-    if(OutputType::DT == L) return "DT";
-    if(OutputType::DC == L) return "DC";
-    if(OutputType::PP == L) return "PP";
-    if(OutputType::VF == L) return "VF";
-    if(OutputType::HIST == L) return "HIST";
-    if(OutputType::LITR == L) return "LITR";
-    if(OutputType::YF == L) return "YF";
-    if(OutputType::BEAME == L) return "BEAME";
-    if(OutputType::BEAMS == L) return "BEAMS";
-    if(OutputType::AMP == L) return "AMP";
-    if(OutputType::NL == L) return "NL";
-
-    return "NL";
+    std::string sInput = to_name(L);
+    std::string Out = std::regex_replace(sInput, std::regex(R"([0-9])"), "");
+    if (Out == "UR") return "U";
```

Comment:
Sorry I somehow did not see this. Please ignore the previous comment.

Reply:
I tried to ensure that all the special cases are handled. Sorry if I missed some.